## Chi-Square Test for Independence
### Suppose a survey was conducted to determine if customer satisfaction (Satisfied/Not
Satisfied) is independent of product type (A/B). The data collected is shown below:

```
				  Product A	Product B	Total
Satisfied		40			30			70
Not Satisfied	20			10			30
Total			60			40			100
```



In [1]:
import numpy as np
from scipy.stats import chi2

In [22]:
observed = np.array([
    [40,30],
    [20,10]
])

significance_level = 0.05
col_total = observed.sum(axis = 0)
row_total = observed.sum(axis = 1)
grand_total = observed.sum()


exp_val_prod_A = int ((int(row_total[0])* int(col_total[0]) )/grand_total)
exp_val_prod_B = int ((int(row_total[0])* int(col_total[1]) )/grand_total)

un_exp_val_prod_A = int ((int(row_total[1]) * int(col_total[0]))/grand_total)
un_exp_val_prod_B = int ((int(row_total[1]) * int(col_total[1]))/grand_total)

col_total, row_total, grand_total, exp_val_prod_A, un_exp_val_prod_A,un_exp_val_prod_B

 # x2 = (40 - 42)2/42 + (30 - 28)2/28 + (20 - 18)2/18 + (10 - 12)2/1
chi_square = float((observed[0][0])- exp_val_prod_A) ** 2/ exp_val_prod_A
+ (int(observed[0][1] - exp_val_prod_B) ** 2/exp_val_prod_B)
+ (int(observed[1][0] - un_exp_val_prod_A) ** 2/un_exp_val_prod_A)
+ (int(observed[1][1] - un_exp_val_prod_B) ** 2/un_exp_val_prod_B)

df= (len(observed) -1 ) * (len(observed[0]) -1 )
critical_value = chi2.ppf(1- significance_level, df)
critical_value


np.float64(3.841458820694124)

####  Since, x2 = 0.793 < 3.841, we fail to reject the null hypothesis.
### Conclusion:
There is no significant relationship between customer satisfaction and product type.

